src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -497,9 +501,19 @@ private MsalAccessTokenCacheItem FilterByExpiry(MsalAccessTokenCacheItem msalAcc
             return null;
         }
 
-        private static MsalAccessTokenCacheItem GetSingleResult(
+        private static MsalAccessTokenCacheItem GetSingleAccessTokenResult(AuthenticationRequestParameters requestParams, IReadOnlyList<MsalAccessTokenCacheItem> filteredItems)
+        {
+            return (MsalAccessTokenCacheItem) GetSingleResult(requestParams, filteredItems);
+        }
+
+        private static MsalRefreshTokenCacheItem GetSingleRefreshTokenResult(AuthenticationRequestParameters requestParams, IReadOnlyList<MsalRefreshTokenCacheItem> filteredItems)
+        {
+            return (MsalRefreshTokenCacheItem) GetSingleResult(requestParams, filteredItems);
```

Comment:
I think you're introducing a change here that should not be introduced. If 2 AT are found by filtering, we throw because there's smth wrong in MSAL. If 2 RTs are found, we just pick one, we don't throw.

Reply:
we shouldn't have two RTs anyway, at least not in the partitioned cache. We have to be really careful just using one of them, as the claims could be different.

Reply:
That was the flow before where we returned the first or default. But I am not sure in which case will we have more than 1 RT with the filters. And if there are more than 1 RTs then will using any one of those be fine?